### PR TITLE
Add table and repo for reconciliation reports

### PR DESCRIPTION
**Pull request by rkondratowicz**
https://tools.hmcts.net/jira/browse/BPS-1358

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/blobrouter/data/reconciliation/ReconciliationReportRowMapper.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            rs.getString("content_type_version"),
+            toDateTime(rs.getTimestamp("sent_at")),
+            toDateTime(rs.getTimestamp("created_at"))
+        );
+    }
+
+    private LocalDateTime toDateTime(Timestamp timestamp) {
+        return timestamp == null
+            ? null
+            : timestamp.toLocalDateTime();
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/data/reconciliation/ReconciliationReportRowMapper.java:32]: not too much info, shall we inline?

File: src/main/java/uk/gov/hmcts/reform/blobrouter/data/reconciliation/ReconciliationReportRepository.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                new MapSqlParameterSource("id", id),
+                this.rowMapper
+            );
+            return Optional.of(report);
+        } catch (EmptyResultDataAccessException ex) {
+            return Optional.empty();
+        }
+    }
+
+    private PGobject toJson(String string) throws SQLException {
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/data/reconciliation/ReconciliationReportRepository.java:66]: argument name string ?

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/data/reconciliation/ReconciliationReportRepository.java:66]: I'm open to all proposals :)

Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/data/reconciliation/ReconciliationReportRepository.java:66]: reportContent may be ?

**Pull request merged**